### DISCUSSION TITLE
How to run Find top markers based on predicated.labels?

### DISCUSSION THREAD

**Discussion by ahmadsam66** [stuart-lab/signac#685] [Q&A] [answered]
Hello

I have used Snare-Seq https://satijalab.org/signac/articles/snareseq.html .

When I tried to run FindAllMarkers for "peaks" or "RNA" it would not work based on predicted.labels.

How can I assign those predicted labels to their cells and run FindAllMarkers based on these labels ?

thanks.

**Comment by timoast** [comment, marked as answer]
You need to add the label predictions using the `AddMetaData()` function (see https://satijalab.org/signac/articles/pbmc_vignette.html#integrating-with-scrna-seq-data for example) and set the `group.by` parameter in `FindAllMarkers()` to `predicted.id`

**Comment by ahmadsam66** [comment]
Thanks @timoast  it works. I have used this code :

```
set.seed(1234)
tumor_atac <- AddMetaData(object = tumor_atac, metadata = predicted.labels)
DefaultAssay(tumor_atac) <- 'peaks'
da.peaks <- FindMarkers(tumor_atac, min.pct = 0.2, test.use = "LR", latent.vars = "peak_region_fragments",only.pos = T , group.by="predicted.id",ident.1="Cluster_1")
```

based on this tutorial:
https://satijalab.org/signac/articles/pbmc_vignette.html#integrating-with-scrna-seq-data
